pkg/agent/core/ngt/handler/grpc/option.go

```diff
@@ -41,40 +43,57 @@ var defaultOptions = []Option{
 	WithErrGroup(errgroup.Get()),
 }
 
+// WithIP returns the option to set the IP for server.
 func WithIP(ip string) Option {
-	return func(s *server) {
-		if len(ip) != 0 {
-			s.ip = ip
+	return func(s *server) error {
+		if len(ip) == 0 {
+			return errors.NewErrInvalidOption("ip", ip)
 		}
+		s.ip = ip
+		return nil
 	}
 }
 
+// WithName returns the option to set the name for server.
 func WithName(name string) Option {
-	return func(s *server) {
-		if len(name) != 0 {
-			s.name = name
+	return func(s *server) error {
+		if len(name) == 0 {
+			return errors.NewErrInvalidOption("name", name)
 		}
+		s.name = name
+		return nil
 	}
 }
 
+// WithNGT returns the option to set the NGT service for server.
 func WithNGT(n service.NGT) Option {
-	return func(s *server) {
+	return func(s *server) error {
+		if n == nil {
+			return errors.NewErrInvalidOption("ngt", n)
+		}
 		s.ngt = n
+		return nil
 	}
 }
 
+// WithStreamConcurrency returns the option to set the stream concurrency for server.
 func WithStreamConcurrency(c int) Option {
-	return func(s *server) {
-		if c != 0 {
-			s.streamConcurrency = c
+	return func(s *server) error {
+		if c <= 0 {
```

Comment:
@vankichi  I found that this variable will be use here.
https://github.com/vdaas/vald/blob/master/internal/net/grpc/stream.go#L54

Since it will not process when c<= 0, I think it is better to check <=0 instead of ==0 here, what do you think?